app/forms/flood_risk_engine/steps/base_address_form.rb

```diff
@@ -32,14 +32,18 @@ def save
 
       # read only param for displaying the Postcode in the view
       def postcode
-        return enrollment.address_search.postcode if enrollment.address_search.present?
+        address_search_postcode || primary_address_postcode || ""
+      end
 
-        return enrollment.organisation.primary_address.postcode if enrollment.organisation.primary_address.present?
+      protected
 
-        ""
+      def address_search_postcode
+        enrollment.address_search.try(:postcode)
```

Comment:
try is ugly.  your delegate style would be better leading to `organisation.primary_address_postcode`